sled-agent/src/instance.rs

```diff
@@ -676,11 +676,22 @@ impl InstanceRunner {
     ) -> Result<(), Error> {
         let nics = running_zone
             .opte_ports()
-            .map(|port| propolis_client::types::NetworkInterfaceRequest {
-                name: port.name().to_string(),
-                slot: propolis_client::types::Slot(port.slot()),
+            .filter_map(|port| {
+                let interface_id = self
+                    .requested_nics
+                    .iter()
+                    // We expect to match NIC slots to OPTE port slots.
+                    .find(|nic| nic.slot == port.slot())
```

Comment:
There should always be one port for each guest NIC. Should we do something here if that's not the case?

Reply:
From talking to others, this should work. I guess the question is if we should error if it's empty altogether?

Reply:
@bnaecker after thinking about it, and knowing that a requested nic should match to a port, I've updated to throw an error otherwise. Thoughts?

Reply:
That seems good!